Marlin/Marlin_main.cpp

```diff
@@ -3298,6 +3410,45 @@ inline void gcode_G28() {
         } //xProbe
       } //yProbe
 
+      #if ENABLED(PROBE_FAIL_PANIC)
+        if(reprobe_attempts == NUM_ATTEMPTS && probePointCounter == -1)
+        {
+          if(!IS_SD_PRINTING)
+            probe_fail = true;
+            disable_all_heaters();
+          #if ENABLED(THERMAL_RUNAWAY_PROTECTION_PERIOD)
+            for(int i=0; i<EXTRUDERS; i++) target_temp_reached[i] = false;
+          #endif
+          #if ENABLED(SDSUPPORT)
+            card.closefile();
+            card.sdprinting = false;
+          #endif
+          clear_buffer();
+          reprobe_attempts = 0;
+          do_blocking_move_to_z(Z_RETRY_PT);
+          // Move E back to 0 after a failed probe
+          plan_buffer_line(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], 0.0, feedrate/60, active_extruder);
```

Comment:
@thinkyhead [L3426](https://github.com/MarlinFirmware/Marlin/pull/3553/files#diff-1cb08de130a6ece2d1b5b9c37bcfef48R3426) (of the most recent commit, same as L3430 here) is intended only if wiping is necessary for probing since a retraction is done prior to wiping, so I wrap it with `#if ENABLED(REWIPE)` in my latest commit.